src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java

```diff
@@ -171,7 +167,11 @@ private WinConnection connectToWinRM(EC2Computer computer, EC2AbstractSlave node
                             Thread.sleep(sleepBetweenAttempts);
                             continue;
                         }
-                        String password = node.getCloud().getPrivateKey().decryptWindowsPassword(passwordData);
+                        EC2PrivateKey ec2PrivateKey = node.getCloud().resolvePrivateKey();
+                        String password = "";
+                        if (ec2PrivateKey != null){
```

Comment:
If I have to decrypt the password but there is no private key is there any point to continuing?

Reply:
I was not sure what other way to go, but continuing with the empty password will most definitely fail which should happen anyways in the case of a missing private key. Any suggestion for better error handling here?

Reply:
We should probably log it and throw an exception. I believe it will terminate the instance in this case

Reply:
A few lines above the missing passwordData is handled by calling `continue` in the whole connection loop with the expectation that the passwordData will be added between the loops. 
Do you think it would lead to other issues handling a null privateKey exactly like that, expecting that the logs are read an the user configured the right credential?

```suggestion
                        if (ec2PrivateKey == null){
                            logger.println("Waiting for privateKey to be available. Consider checking the credentials in the cloud configuration. Sleeping 10s. ");
                            Thread.sleep(sleepBetweenAttempts);
                            continue;
                        }
                        String password = ec2PrivateKey.decryptWindowsPassword(passwordData);
```

Seems for me like a legit way to handle this as nothing gets terminated but a correct configuration gets enforced.

Reply:
I now made this change as it makes sense to me like that. Thanks for pointing that out